lib/between_meals/changes/cookbook.rb

```diff
@@ -44,19 +38,59 @@ def self.explode_path(path, cookbook_dirs)
         nil
       end
 
+      def self.symlinked_dir?(link_path)
+        File.symlink?(link_path) &&
+          File.directory?(File.absolute_path(link_path))
+      end
+
+      def self.map_symlinks(files)
+        # For each symlink get the source path, if any files have changed under
+        # the source path, fake them as coming from the symlink path. This
+        # allows the normal cookbook logic to just work.
+        require 'find'
+        symlinks = {}
+        @cookbook_dirs.each do |dir|
+          dir = File.join(@repo_dir, dir)
+          # Finds all symlinks in cookbook dirs
+          Find.find(dir).select { |f| f if File.symlink?(f) }.each do |link|
+            next if symlinks[link]
+            source = File.absolute_path(link)
+            repo = File.join(@repo_dir, '/')
+            # maps absolute symlink path to relative source and link paths
+            symlinks[link] = {
+              'source' => source.gsub(repo, ''),
+              'link' => link.gsub(repo, ''),
+            }
+          end
+        end
+
+        # Create the data hash expected for each file but fake the source path
```

Comment:
Then this block walks thie list of symlinks to find if any paths match them and the munges the data structure to fake the path as described.